src/solvers/decision_procedure.cpp

```diff
@@ -11,13 +11,21 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include "decision_procedure.h"
 
+#include <util/std_expr.h>
+
 decision_proceduret::~decision_proceduret()
 {
 }
 
 decision_proceduret::resultt decision_proceduret::operator()()
 {
-  return dec_solve();
+  return dec_solve(nil_exprt());
+}
```

Comment:
My preference would be to use std::optional at the interface rather than nil.

Reply:
I was reluctant to do so since that can't avoid copying the expression.

Reply:
Wouldn't `const std::optional<exprt> &` do the trick?

Reply:
No, you'd still need to copy the expression into the optional when creating the optional.

Reply:
`std::optional< std::reference_wrapper<exprt> >` would work, but then it's getting more verbose.